### PR TITLE
Add NiFi Registry Deployment

### PR DESCRIPTION
**Pull request by jasonrogena**
Adds the NiFi Registry deployment [in build.openlmis.org](http://build.openlmis.org/job/NiFi-Registry-deploy-to-shared/).

### CODE REVIEW

**Comment by craigappl** [comment]
I added @joshzamor  as a reviewer. 1) I think it's a great opportunity for him to see how this was done and 2) I don't have the technical knowledge to assess the process.

**Comment by craigappl** [comment]
I added @joshzamor  as a reviewer. 1) I think it's a great opportunity for him to see how this was done and 2) I don't have the technical knowledge to assess the process.

**Comment by jasonrogena** [comment]
Cool. I'll go ahead and add the fixes for https://openlmis.atlassian.net/browse/OLMIS-5231 here too, then.

**Comment by jasonrogena** [comment]
@joshzamor I've added support for setting up Docker and Docker Compose in the uat Terraform deployments. Could you please test those for me 🙏. The Ansible playbook that'll run will create new TLS certs for Docker and update the daemon, so the Jenkins deployments will fail until you update the keys.

File: provision/terraform/modules/openlmis/variables.tf [hunk 1]
```diff
@@ -1,4 +1,39 @@
...
+  type        = "string"
+  description = "The path to the directory containing the playbook for installing docker"
+}
+
+variable "docker-tls-port" {
+  type        = "string"
+  description = "The TCP port The Docker Daemon is listening for TLS traffic"
+}
+
+variable "app-tls-s3-access-key-id" {
```
Comment (joshzamor) [inline provision/terraform/modules/openlmis/variables.tf:21]: So if I understand this right, Ansible will use the AWS access key to save the tls files to, right?  Is this needed or can we just use the AWS access key of the one running Terraform?

**Comment by joshzamor** [comment]
Thanks for this @jasonrogena.  I've worked through the changes and I believe I have a general understanding of it.  More so on the Terraform side and less on the Ansible side.  I've left a comment but otherwise I'll be attempting to use this to launch a new OpenLMIS demo.

File: provision/terraform/reporting/nifi-registry/terraform.tfvars [hunk 2]
```diff
@@ -0,0 +1,16 @@
+nr-instance-type = "t2.medium"
+nr-volume-size = "200"
+nr-ssh-key = "ona-terraform-tbd"
```
Comment (joshzamor) [inline provision/terraform/reporting/nifi-registry/terraform.tfvars]: Could we make sure that this AWS key is in our S3 bucket?

**Review by joshzamor — CHANGES_REQUESTED** [review]
@jasonrogena I was almost able to use this however I hit a snag in Ansible, and I think it's simply my misunderstanding of what I need to do to install the ansible prereqs correctly.  I started `provision/terraform/README.md` and it'd be great if you could add some high-level steps and requirements for making the whole thing work (e.g. do I need to go to `provision/ansible` and follow the steps in the `README.md` there to install Ansible prereqs?

File: provision/terraform/modules/openlmis/variables.tf (hunk 1 above)
Comment (jasonrogena) [reply provision/terraform/modules/openlmis/variables.tf:21]: Yeah, that's true, though we'd need to find a way to do that. Terraform doesn't expose the access key it uses. One way we could possibly do this is through Terraform [environment variables](https://www.terraform.io/docs/configuration/environment-variables.html):

```sh
export TF_VAR_app-tls-s3-access-key-id=$AWS_ACCESS_KEY_ID
export TF_VAR_app-tls-s3-secret-access-key=$AWS_SECRET_ACCESS_KEY
```

This however assumes `AWS_SECRET_ACCESS_KEY` and `AWS_ACCESS_KEY_ID` have been set (and Terraform isn't using the `~/.aws/credentials` file).

File: provision/terraform/reporting/nifi-registry/terraform.tfvars (hunk 2 above)
Comment (jasonrogena) [reply provision/terraform/reporting/nifi-registry/terraform.tfvars]: Done. I've changed the key to `openlmis-reporting`. I've copied the private key to the S3 bucket.

File: provision/terraform/modules/openlmis/variables.tf (hunk 1 above)
Comment (jasonrogena) [reply provision/terraform/modules/openlmis/variables.tf:21]: Added this to the setup instructions.

**Review by joshzamor — APPROVED** [review]

File: provision/ansible/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+```sh
+pip install -r requirements/ansible.pip
+```
+
+Install the [Ansible Galaxy](https://galaxy.ansible.com/docs/) dependencies:
+
+```sh
+mkdir -p vendor/roles
+ansible-galaxy install -p vendor/roles -r requirements/galaxy.yml
```
Comment (joshzamor) [inline provision/ansible/README.md:15]: @jasonrogena :  When I try to run this on my machine (OS X.11) I'm getting the following error:

```
ansible-galaxy install -p vendor/roles -r requirements/galaxy.yml
Traceback (most recent call last):
  File "/usr/local/bin/ansible-galaxy", line 67, in <module>
    import ansible.constants as C
  File "/Users/Josh-LT/Library/Python/2.7/lib/python/site-packages/ansible/constants.py", line 11, in <module>
    from jinja2 import Template
ImportError: No module named jinja2
```

I ran `sudo pip install jinja2`, which succeeds, however when I run the ansible command again I get the same error.  I'm not using virtualenv which is not the best idea, however I'm wondering if we're missing something.  Perhaps we can have another dev try this with their machine and see if it works?

**Review by joshzamor — CHANGES_REQUESTED** [review]

**Review by joshzamor — APPROVED** [review]

**Comment by joshzamor** [comment]
@jasonrogena :  I'm still having trouble with Ansible on my machine however I'm afraid this PR will get stale here so I'm going to merge it.  Thanks for the great work here, and if you do have someone else try out the Ansible stuff on their machine I'd be glad to know of it.  I'll keep hacking away on my machine to figure out what's wrong.

**Pull request merged**